### PR TITLE
tests: add spread test for registry cross-snap config

### PR DESCRIPTION
**Pull request by MiguelPires**
Adds a spread test that tests cross-snap configuration in registries, including checking that the correct hooks are called for custodian and non-custodian snaps and that failed save-view hooks are rolled back. This is based on https://github.com/canonical/snapd/pull/14620 so only commits e50e0b2 and onwards are relevant

### CODE REVIEW

**Comment by MiguelPires** [comment]
Closing as this has been split/reworked into several other PRs

File: overlord/registrystate/registrymgr.go [hunk 1]
```diff
@@ -204,6 +204,7 @@ func (h *saveViewHandler) Error(origErr error) (ignoreErr bool, err error) {
 	// create roll back tasks for the previously done save-registry hooks (starting
 	// with the hook that failed, so it tries to overwrite with a pristine databag
 	// just like any previous save-view hooks)
+	haltTasks := t.HaltTasks()
```
Comment (andrewphelpsj) [inline overlord/registrystate/registrymgr.go:207]: Is there some ordering issue here? If so probably worth a comment.

**Review by andrewphelpsj — APPROVED** [review]
Thank you!

**Review by ZeyadYasser — APPROVED** [review]
Thank you!

**Comment by MiguelPires** [comment]
The spread test added in this PR is not failing in any system. The failures are known:

* ubuntu-fips-22.04-64:tests/fips/
* ubuntu-core-18-64:tests/core/snapd-refresh-vs-services:*
* ubuntu-24.04-64:tests/main/upgrade-from-release
* opensuse-tumbleweed-64 (failed suite prepare)
* fedora-40-64:tests/main/component-from-store
* fedora-39-64:tests/main/component-from-store

**Pull request merged**